app/tasks/vector_source_assets.py

```diff
@@ -0,0 +1,150 @@
+    assert len(source_uris) == 1, "Vector sources only support one input file"
+
+    options = VectorSourceCreationOptions(**creation_options)
+
+    # source_uri: str = gdal_path(source_uris[0], options.zipped)
+    source_uri = source_uris[0]
+    local_file = os.path.basename(source_uri)
+
+    if options.layers:
+        layers = options.layers
+    else:
+        layer, _ = os.path.splitext(os.path.basename(source_uri))
+        layers = [layer]
+
+    data = AssetTaskCreate(
+        asset_type="Database table",
+        dataset=dataset,
+        version=version,
+        asset_uri=f"/{dataset}/{version}/features",
+        is_managed=True,
+        creation_options=options,
+        metadata=DatabaseTableMetadata(**metadata),
+    )
+
+    async with ContextEngine("PUT"):
+        new_asset = await assets.create_asset(**data.dict())
+
+    create_vector_schema_job = GdalPythonImportJob(
+        job_name="import_vector_data",
+        command=[
+            "create_vector_schema.sh",
+            "-d",
+            dataset,
+            "-v",
+            version,
+            "-s",
+            source_uri,
+            "-l",
+            layers[0],
+            "-f",
+            local_file,
+        ],
+        environment=writer_secrets,
+    )
+
+    load_vector_data_jobs: List[Job] = list()
+    for layer in layers:
+        load_vector_data_jobs.append(
+            GdalPythonImportJob(
+                job_name="load_vector_data",
+                command=[
+                    "load_vector_data.sh",
+                    "-d",
+                    dataset,
+                    "-v",
+                    version,
+                    "-s",
+                    source_uri,
+                    "-l",
+                    layer,
+                    "-f",
+                    local_file,
+                ],
+                parents=[create_vector_schema_job.job_name],
+                environment=writer_secrets,
+            )
+        )
+
+    gfw_attribute_job = PostgresqlClientJob(
+        job_name="enrich_gfw_attributes",
+        command=["add_gfw_fields.sh", "-d", dataset, "-v", version],
+        parents=[job.job_name for job in load_vector_data_jobs],
+        environment=writer_secrets,
+    )
+
+    index_jobs: List[Job] = list()
+
+    for index in options.indices:
+        index_jobs.append(
+            PostgresqlClientJob(
+                job_name=f"create_index_{index.column_name}_{index.index_type}",
+                command=[
+                    "create_index.sh",
+                    "-d",
+                    dataset,
+                    "-v",
+                    version,
+                    "-c",
+                    index.column_name,
+                    "-x",
+                    index.index_type,
+                ],
+                parents=[gfw_attribute_job.job_name],
+                environment=writer_secrets,
+            )
+        )
+
+    inherit_geostore_job = PostgresqlClientJob(
```

Comment:
Should this pipeline also check for partitions? I thought you partition the fire vector data